### ISSUE TITLE
Regression in 1.9.11 -> 1.10.1 : line coverage stats >100%

### ISSUE THREAD

**Issue by therealryan** [hcoles/pitest#1126]
Dependabot wants to update [my project](https://github.com/Mastercard/flow) from 1.9.11 to 1.10.1, but [the PR](https://github.com/Mastercard/flow/pull/151) is failing as line coverage on the first subproject has dropped from 100% to 98% according to the [build output](https://github.com/Mastercard/flow/actions/runs/3615839053/jobs/6093245051).

When I try to recreate the failure locally the generated report shows an overall line coverage stat of **106%**, and even **133%** in one class. Predictably enough, the >100% coverage class reports show no uncovered lines and no obvious weirdness.
I think my tests are pretty good, but they're not _that_ good.

Hopefully others can recreate this result by:
 1. `git clone https://github.com/Mastercard/flow.git && cd flow`
 2. `mkdir -p api/target && touch api/target/mutate`
 4. `mvn test-compile`

Cheers!

**Comment by hcoles** [comment]
Thanks for reporting.

This is most likely fixed by #1125, which will be released as 1.10.2 shortly. If not, let me know and I'll take a look.

**Comment by hcoles** [comment]
@therealryan 1.10.2 has now been released. It should be available from maven central in the next hour (but won't show up in search sites/dependabot for a day or so).

**Comment by therealryan** [comment]
It seems maven-central is feeling snappy today! `1.10.2` has _changed_ the results but not really improved matters.
Now the build output reports line coverage of 84% (so the build is failed), but the generated report shows 100% (so I can't see what the issue is).

**Comment by hcoles** [comment]
@therealryan Ok, looks like something was badly messed up with the thresholds during the 1.9 -> 1.10 refactor, which is a little embarassing. Thanks again for reporting, hopefully this was stupidity confined to a single place rather than something wide spread.

**Comment by hcoles** [comment]
Fortunately looks to be a simple issue. Lines are being double counted when generating the threshold if they're included in the bytecode more than once. The compiler would do this for finally blocks, and perhaps in some other scenarios.

It also seems that the old html report code is calculating this stuff seperately, which seems less than ideal.

Will poke around for a bit to see if anything else looks broken, then produce a release shortly.

**Comment by hcoles** [comment]
Couldn't see any additional issues, so release 1.10.3 is now syncing. Hopefully there's nothing else lurking.

**Comment by therealryan** [comment]
Yep, [looking good with 1.10.3](https://github.com/Mastercard/flow/actions/runs/3620424979). Thanks!
